src/module_3/Module3_Milestone1.md

```diff
@@ -0,0 +1,1825 @@
+
+train_val_cut = cum_sum_daily_orders[cum_sum_daily_orders <= 0.7].idxmax()
+val_test_cut = cum_sum_daily_orders[cum_sum_daily_orders <= 0.9].idxmax()
+
+print("Train set from: ", cum_sum_daily_orders.index.min())
+print("Train set until: ", train_val_cut)
+print("Validation set from: ", val_test_cut)
+print("Test set until: ", cum_sum_daily_orders.index.max())
+```
+
+    Train set from:  2020-10-05 00:00:00
+    Train set until:  2021-02-04 00:00:00
+    Validation set from:  2021-02-22 00:00:00
+    Test set until:  2021-03-03 00:00:00
+
+
+It is important to consider what is the minimum business cicle, that is, the minimum temporal window that allows us to capture the complete dynamics of the data we want to model. Even though the validation and test splits don't comprise many days, we consider these should be sufficient for our case.
+
+
+```python
+train_data = filtered_data[filtered_data.order_date <= train_val_cut]
+val_data = filtered_data[(filtered_data.order_date > train_val_cut) & (filtered_data.order_date <= val_test_cut) ]
+test_data = filtered_data[filtered_data.order_date > val_test_cut]
+```
+
+Divide into features and target:
+
+
+```python
+def feature_target_split(df, target):
+    X = df.drop(columns=[target])  # Drop the target column to create the features DataFrame
+    Y = df[target]   # Target variable we want to predict
+    return X, Y
+
+X_train, Y_train = feature_target_split(train_data, target_col)
+X_val, Y_val = feature_target_split(val_data, target_col)
+X_test, Y_test = feature_target_split(test_data, target_col)
+```
+
+### 2. Defining a Baseline model
+
+In order to be able to decide if a ML model generates any value, we need to compare it against baselines that do not require training. In this case, we will use the global popularity feature as baseline, that is, the product will be predicted to be bought the more popular it is.
+
+
+```python
+def plot_curves(y_true, y_pred, curve_type="both", dataset_type="train"):
+    
+    fig, axes = plt.subplots(1, 2, figsize=(14, 6))
+    fig.suptitle(f"{dataset_type.capitalize()} Dataset - Performance Curves")
+
+    if curve_type in ["precision-recall", "both"]:
+        # Compute the precision-recall curve and its AUC
+        precision, recall, _ = precision_recall_curve(y_true, y_pred)
+        pr_auc = auc(recall, precision)
+        
+        axes[0].step(recall, precision)
+        axes[0].set_xlabel('Recall')
+        axes[0].set_ylabel('Precision')
+        axes[0].set_title(f'Precision-Recall Curve (AUC={pr_auc:.2f})')
+
+    if curve_type in ["roc", "both"]:
+        # Compute the ROC curve and its AUC
+        fpr, tpr, _ = roc_curve(y_true, y_pred)
+        roc_auc = auc(fpr, tpr)
+
+        axes[1].plot(fpr, tpr)
+        axes[1].set_xlabel('False Positive Rate')
+        axes[1].set_ylabel('True Positive Rate')
+        axes[1].set_title(f'ROC Curve (AUC={roc_auc:.2f})')
+
+    plt.tight_layout()
```

Comment:
This is fine, but a general tip: consider returning the axes so that external methods can leverage this plot and build on top. OF course after consieration you may want to leave it like this